### ISSUE TITLE
Would it be possible for racer to integrate with rustdoc?

### ISSUE THREAD

**Issue by 7sharp9** [phildawes/racer#415]

**Comment by bruno-medeiros** [comment]
+1 , would be a great feature.

**Comment by phildawes** [comment]
Agreed. Currently all my spare time is going into figuring out how to use rustc to get perfect completions so I probably won't get to this for a while I'm afraid.

**Comment by estebank** [comment]
I have a small hacked together version of this, I'll be sending a pull request later today.

I noticed that for some reason in HEAD `complete-with-snippets` is returning the same thing as `complete`. I'll try to get it working with vim-racer as well to present a full feature.

I modified `core::Match` to add a `docs: String` field. Let me know if you disagree with doing that for some reason, @phildawes.
